### PR TITLE
Display Mailbox for tablet

### PR DESCRIPTION
**Pull request by dab246**
- [x] Display two-screen navigation on mobile devices
- [x] Display two-part interface on tablets.
- [x] Display three-part interface on desktop.
- [ ] Minimize - expand mailbox menu on tablets

https://user-images.githubusercontent.com/80730648/127309712-eba67d37-aba0-46b8-9e97-5c3a793bcd82.mov

### CODE REVIEW

**Comment by chibenwa** [comment]
Looks good to be honnest!

Two tiny remarks:

 - On the tablet veiw we can likely still display quota information (we have room for it). I agree that mobile landscape view not to display it, as space is more scarce.
 
  - We you open a mailbox the count do not correspond to the count of the mailbox list ;-)
  
Cheers,

Benoit

**Comment by hoangdat** [comment]
> We you open a mailbox the count do not correspond to the count of the mailbox list ;-)

Srr, I can not get your point. you mean `the count`? what is this?

**Comment by dab246** [comment]
> > We you open a mailbox the count do not correspond to the count of the mailbox list ;-)
> 
> Srr, I can not get your point. you mean `the count`? what is this?

He means the value `1 new` in the mail list screen. Here we just build the interface. So all figures may not be accurate because I faked it.

File: lib/features/messenger/presentation/messenger_view.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_listmail/presentation/mailbox_listmail_controller.dart';
+import 'package:tmail_ui_user/features/messenger/presentation/messenger_controller.dart';
+import 'package:tmail_ui_user/features/messenger/presentation/widgets/app_bar_messenger_widget_builder.dart';
+import 'package:tmail_ui_user/features/messenger/presentation/widgets/bottom_bar_messenger_widget_builder.dart';
+
+class MessengerView extends GetWidget<MessengerController> {
```
Comment (hoangdat) [inline lib/features/messenger/presentation/messenger_view.dart:11]: `MailView`

File: lib/features/mailbox_listmail/presentation/mailbox_listmail_view.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/mailbox_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_listmail/presentation/mailbox_listmail_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_listmail/presentation/widgets/app_bar_listmail_widget_builder.dart';
+import 'package:tmail_ui_user/features/mailbox_listmail/presentation/widgets/mail_tile_builder.dart';
+
+class MailboxListMailView extends GetWidget<MailboxListMailController> {
```
Comment (hoangdat) [inline lib/features/mailbox_listmail/presentation/mailbox_listmail_view.dart:12]: `ThreadView`

**Comment by chibenwa** [comment]
:+1: I <3 this ;-)

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 3]
```diff
@@ -54,6 +64,38 @@ class MailboxController extends BaseController {
...
 
+  void _setPresentationMailboxDefault(List<PresentationMailbox> defaultMailboxList) {
+    try {
+      final mailboxDefault = defaultMailboxList
+          .firstWhere((presentationMailbox) => presentationMailbox.role == Role(MailboxConstants.ROLE_DEFAULT));
+      presentationMailboxCurrent.value = mailboxDefault;
+    } catch (e) {}
+  }
+
+  SelectMode getSelectModeMailboxCurrent(PresentationMailbox presentationMailbox) {
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:75]: `getSelectMode`

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 4]
```diff
@@ -54,6 +64,38 @@ class MailboxController extends BaseController {
     folderMailboxTree.value = await _treeBuilder.generateMailboxTree(folderMailboxList);
   }
 
+  void _setPresentationMailboxDefault(List<PresentationMailbox> defaultMailboxList) {
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:67]: `_setDefaultSelected`

File: lib/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:get/get.dart';
+
+class MailboxDashBoardController extends GetxController {
+
+  final scaffoldKey = GlobalKey<ScaffoldState>();
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart:7]: In order for us to be able to access the MailboxDashBoardView widget, we need to open and close the drawer

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**